teuthology/lock.py

```diff
@@ -226,6 +226,13 @@ def main(ctx):
     if ctx.num_to_lock:
         assert ctx.machine_type, \
             'must specify machine type to lock'
+    if ctx.machine_type == 'vps':
+        if ctx.os_type is None:
+            ctx.os_type=random.choice(default_table.keys())
+	    ctx.os_version=random.choice(default_table[ctx.os_type])
```

Comment:
What you're doing here is adding an unrelated, unexpected side-effect to a function designed to simply "Return a table of valid distros". It is not the place where decisions about *what* distro to use should be made.

It is currently made here for downburst:
https://github.com/ceph/teuthology/blob/master/teuthology/provision.py#L210-L211

To properly implement this would require a little bit of refactoring I think.